diff --git a/fftools/ffmpeg_dec.c b/fftools/ffmpeg_dec.c
index fcee8b65ac..3b27e15279 100644
--- a/fftools/ffmpeg_dec.c
+++ b/fftools/ffmpeg_dec.c
@@ -215,6 +215,9 @@ static void audio_ts_process(void *logctx, Decoder *d, AVFrame *frame)
     // generation that can represent timestamps from all the samplerates
     // seen so far
     tb = audio_samplerate_update(logctx, d, frame);
+    av_log(logctx, AV_LOG_WARNING,
+                "Remotion Repro: tb: %d sr: %d\n", tb, frame->sample_rate);
+
     pts_pred = d->last_frame_pts == AV_NOPTS_VALUE ? 0 :
                d->last_frame_pts + d->last_frame_duration_est;
 
diff --git a/libavcodec/proresenc_kostya.c b/libavcodec/proresenc_kostya.c
index 52fe5639b1..33655d5810 100644
--- a/libavcodec/proresenc_kostya.c
+++ b/libavcodec/proresenc_kostya.c
@@ -562,8 +562,7 @@ static void encode_alpha_plane(ProresContext *ctx, PutBitContext *pb,
             run++;
         }
     } while (idx < num_coeffs);
-    if (run)
-        put_alpha_run(pb, run);
+    put_alpha_run(pb, run);
 }
 
 static int encode_slice(AVCodecContext *avctx, const AVFrame *pic,
