diff --git a/libavcodec/codec_desc.c b/libavcodec/codec_desc.c
index 03dea5751a..f2aece7566 100644
--- a/libavcodec/codec_desc.c
+++ b/libavcodec/codec_desc.c
@@ -1274,9 +1274,9 @@ static const AVCodecDescriptor codec_descriptors[] = {
         .id        = AV_CODEC_ID_HEVC,
         .type      = AVMEDIA_TYPE_VIDEO,
         .name      = "hevc",
-        .long_name = NULL_IF_CONFIG_SMALL("H.265 / HEVC (High Efficiency Video Coding)"),
+        .long_name = "H.265 / HEVC (High Efficiency Video Coding)",
         .props     = AV_CODEC_PROP_LOSSY | AV_CODEC_PROP_REORDER,
-        .profiles  = NULL_IF_CONFIG_SMALL(ff_hevc_profiles),
+        .profiles  = ff_hevc_profiles,
     },
     {
         .id        = AV_CODEC_ID_FIC,
diff --git a/libavcodec/hevc/hevcdec.c b/libavcodec/hevc/hevcdec.c
index 0dc24f82f8..abbb949aef 100644
--- a/libavcodec/hevc/hevcdec.c
+++ b/libavcodec/hevc/hevcdec.c
@@ -4079,7 +4079,7 @@ const FFCodec ff_hevc_decoder = {
     .caps_internal         = FF_CODEC_CAP_EXPORTS_CROPPING |
                              FF_CODEC_CAP_USES_PROGRESSFRAMES |
                              FF_CODEC_CAP_INIT_CLEANUP,
-    .p.profiles            = NULL_IF_CONFIG_SMALL(ff_hevc_profiles),
+    .p.profiles            = ff_hevc_profiles,
     .hw_configs            = (const AVCodecHWConfigInternal *const []) {
 #if CONFIG_HEVC_DXVA2_HWACCEL
                                HWACCEL_DXVA2(hevc),
diff --git a/libavcodec/hevc/ps.c b/libavcodec/hevc/ps.c
index f18b88489b..2cd4d24c5a 100644
--- a/libavcodec/hevc/ps.c
+++ b/libavcodec/hevc/ps.c
@@ -248,13 +248,11 @@ static int decode_profile_tier_level(GetBitContext *gb, AVCodecContext *avctx,
     ptl->tier_flag     = get_bits1(gb);
     ptl->profile_idc   = get_bits(gb, 5);
 
-#if !CONFIG_SMALL
     for (int i = 0; ff_hevc_profiles[i].profile != AV_PROFILE_UNKNOWN; i++)
         if (ff_hevc_profiles[i].profile == ptl->profile_idc) {
             profile_name = ff_hevc_profiles[i].name;
             break;
         }
-#endif
     av_log(avctx, profile_name ? AV_LOG_DEBUG : AV_LOG_WARNING,
            "%s profile bitstream\n", profile_name ? profile_name : "Unknown");
 
diff --git a/libavcodec/profiles.c b/libavcodec/profiles.c
index 3cef82be3b..d7b10490de 100644
--- a/libavcodec/profiles.c
+++ b/libavcodec/profiles.c
@@ -22,7 +22,6 @@
 #include "defs.h"
 #include "profiles.h"
 
-#if !CONFIG_SMALL
 
 const AVProfile ff_aac_profiles[] = {
     { AV_PROFILE_AAC_LOW,   "LC"       },
@@ -203,4 +202,3 @@ const AVProfile ff_evc_profiles[] = {
     { AV_PROFILE_UNKNOWN },
 };
 
-#endif /* !CONFIG_SMALL */
